src/config.py

```
@@ -26,4 +27,45 @@
 # Collection Configuration (env override supported)
 COLLECTION_NAME = 'beagleboard'
 
-RAG_BACKEND_URL = 'https://mind-api.beagleboard.org/api'
\ No newline at end of file
+RAG_BACKEND_URL = 'https://mind-api.beagleboard.org/api'
+
+class ConfigManager:
+    def __init__(self, path="~/.beaglemind_config.json"):
+        self.path = os.path.expanduser(path)
```

I just got an idea, there a many arrys in the cli file, we can get these arrays to config file since they are basically backend (api) and llm config selection opions, they belong here in my opinion, 
also by then we can replace the values here by GROQ_MODELS[0] and LLM_BACKENDS[0] for example, and the collection name should obviously take COLLECTION_NAME constant value

---

I completely agree with the suggestion moving the backend and model arrays to the config file is a cleaner design choice since they represent configuration data, not CLI logic. This separation will make the system easier to maintain and scale, and replacing hardcoded defaults with dynamic references like GROQ_MODELS[0] and LLM_BACKENDS[0] will make future updates smoother and less error-prone.

I’ll make sure to incorporate this change along with the other improvements you mentioned.
Additionally, I was thinking we could make the backend and model arrays dynamically loadable from the config file itself, instead of being hardcoded in Python. That way, users can easily extend available backends or models without modifying the CLI code.
Should i do this too ?

Also thanks alot for all the suggestions and going through my code it means alot as a new open source contributor.

---

Sounds like a  good idea, please document it in the PR description and I'll be waiting for you future changes. Thanks for the efforts!